pkg/database/openshiftclusters.go

```diff
@@ -120,6 +123,24 @@ func (c *openShiftClusters) Get(ctx context.Context, key string) (*api.OpenShift
 	}
 }
 
+func (c *openShiftClusters) QueueLength(ctx context.Context) (int, error) {
+	// QueryRAW return *map[interface{}]interface{} for sorting
+	result := c.c.QueryRaw("", &cosmosdb.Query{
+		Query: `SELECT VALUE COUNT(1) FROM OpenShiftClusters doc WHERE NOT (doc.openShiftCluster.properties.provisioningState IN ("Succeeded", "Failed")) AND (doc.leaseExpires ?? 0) < GetCurrentTimestamp() / 1000`,
+	})
+
+	headers := http.Header{}
+	// TODO: This is strange, but without this it does not work
+	headers.Set("x-ms-documentdb-partitionkeyrangeid", "0")
```

Comment:
this doesn't look reliable - what caused you to add this?
I am wondering if `x-ms-documentdb-query-iscontinuationexpected` might help?
https://stackoverflow.com/questions/44590614/documentdb-query-that-expects-full-results-from-aggregate-functions-is-not-sup